doc/source/whatsnew/v0.17.0.txt

```diff
@@ -110,6 +110,7 @@ Other enhancements
 
 - ``pd.merge`` will now allow duplicate column names if they are not merged upon (:issue:`10639`).
 
+- ``msgpack`` submodule updated to 0.4.6 (:issue:`10581`)
 .. _whatsnew_0170.api:
```

Comment:
Yes you would get the output of `pandas.io.packers.encode(obj)`, so a nested dict of some sort.  And if you supply the `object_hook=decode` you can get the original object back.  What is done for pandas objects in `packers` is just supplying `default=encode` arg to `pack` and `object_hook=decode` to `unpack` and other arguments.